contributed_definitions/NXapm_compositionspace_results.nxdl.xml

```diff
@@ -0,0 +1,375 @@
+                             TODO
+                        </doc>
+                        <dimensions rank="1">
+                            <dim index="1" value="i"/>
+                        </dimensions>
+                    </field>
+                </group>
+            </group>
+            <group name="ic_opt" type="NXprocess">
+                <doc>
+                     Information criterion minimization.
+                </doc>
+                <field name="sequence_index" type="NX_POSINT">
+                    <enumeration>
+                        <item value="3"/>
+                    </enumeration>
+                </field>
+                <group name="cluster_analysisID" type="NXobject">
+                    <doc>
+                         TODO
+                    </doc>
+                    <field name="n_ic_cluster" type="NX_UINT" units="NX_UNITLESS">
+                        <doc>
+                             TODO
+                        </doc>
+                    </field>
+                    <field name="y_pred" type="NX_UINT" units="NX_UNITLESS">
+                        <doc>
+                             TODO
+                        </doc>
+                        <dimensions rank="1">
+                            <dim index="1" value="c"/>
+                        </dimensions>
+                    </field>
+                </group>
+            </group>
+        </group>
+        <group name="clustering" type="NXprocess" minOccurs="1" maxOccurs="1">
+            <doc>
+                 Clustering is the third step of the algorithmic chain that the CompositionSpace
+                 tool implements. In this step the different voxelized regions in the dataset with
+                 eventually different chemical nature are analyzed if they are distributed in
+                 connected groups of voxels with the same chemical nature.
+                 Using the discretization grid from the first step as the support, this enables
+                 to prepare for the fourth step. In the fourth step the interfaces between
+                 identified groups of different chemical nature are triangulated.
+                 
+                 These triangulated network of interfaces between regions of different chemical
+                 nature is the final data structure that the CompositionSpace tool yields.
+                 Subsequently, this mesh can be post-processed, similarly like meshes that are
+                 obtained from isosurface-based approaches, to perform line profile analyses.
+            </doc>
+            <field name="sequence_index" type="NX_POSINT">
+                <enumeration>
+                    <item value="3"/>
+                </enumeration>
+            </field>
+            <group name="cluster_analysisID" type="NXobject" minOccurs="0" maxOccurs="unbounded">
+                <field name="epsilon" type="NX_FLOAT" units="NX_LENGTH">
+                    <doc>
+                         TODO
```

Comment:
note the missing documentation

Reply:
This is intentionally missing as it should be filled in by the IUC09 members, I have spent enough time on the implementation for this project right now.